WcaOnRails/app/models/user.rb

```diff
@@ -859,34 +875,45 @@ def editable_fields_of_user(user)
         fields += %i(receive_delegate_reports)
       end
     end
-    if admin? || board_member? || senior_delegate?
-      fields += %i(delegate_status senior_delegate_id region)
+    fields
+  end
+
+  private def editable_competitor_info_fields(user)
+    fields = Set.new
+    if user == self || admin? || any_kind_of_delegate? || results_team? || communication_team?
+      cannot_edit_data = !!cannot_edit_data_reason_html(user)
+      if !cannot_edit_data
```

Comment:
Nit: `unless`. In this case, you might as well retain the `if !condition` and scrap the `condition = !!some_string` part above. Negating once within the `if` clause should coerce the return value of `cannot_edit_data_reason_html` into a boolean automatically.

Personal preference to choose between
1. `unless` (nice) and `!!` (ugly)
2. `if !` (ugly) but no `!!` (nice)

I don't have any strong feelings, do whatever you think looks best.

Reply:
🤦 Ya that's a bit messy and I hadn't realized it (before I was using cannot_edit_data in multiple places, but the refactored to this).

`unless condition` with no `!` works just fine! no need for the double negative anymore, I'm too used to JS where that would potentially be needed.

Reply:
Ugh, that is almost a bit too much conciseness for my taste, but I guess since it's Ruby we'll be fine :sweat_smile: Thanks!